api/routes/products.js

```diff
@@ -1,20 +1,24 @@
-const express = require('express');
-const router = express.Router();
-const categoryData = require('./../../src/data/categories.json');
-const productData = require('./../../src/data/products.json');
-const all_categories = categoryData.categories;
-const master_products = productData.products.map(p => ({...p, categoryName: all_categories.find(c => c.id === p.categoryId).categoryName || ''}));
+const express = require('express')
+const router = express.Router()
+const dataService = require('./../../dataService')
 
-router.get('/all', (req, res, next)=>{
-     res.status(200).json({
-        products : master_products
-    })
+/*
+endpoint for returning combined products
+*/
+router.get('/all', (req, res, next) => {
+  res.status(200)
+  res.json({
+    products: Object.fromEntries(dataService.getCombinedProductMap())
```

Comment:
```suggestion
    products: dataService.getCombinedProductMap()
```
As it is already a object map?

Reply:
Yes, we do because there is an empty response if I don't use it.